packages/discovery/src/discovery/handlers/user/ArbitrumActorsHandler.ts

```diff
@@ -0,0 +1,141 @@
+import { providers, utils } from 'ethers'
+import * as z from 'zod'
+
+import { EthereumAddress } from '../../../utils/EthereumAddress'
+import { Hash256 } from '../../../utils/Hash256'
+import { DiscoveryLogger } from '../../DiscoveryLogger'
+import { DiscoveryProvider } from '../../provider/DiscoveryProvider'
+import { Trace } from '../../provider/TransactionTrace'
+import { ClassicHandler, HandlerResult } from '../Handler'
+
+export type ArbitrumActorsHandlerDefinition = z.infer<
+  typeof ArbitrumActorsHandlerDefinition
+>
+export const ArbitrumActorsHandlerDefinition = z.strictObject({
+  type: z.literal('arbitrumActors'),
+  actorType: z.union([z.literal('validator'), z.literal('batchPoster')]),
+})
+
+export class ArbitrumActorsHandler implements ClassicHandler {
+  readonly dependencies: string[] = []
+  readonly setValidatorFn = 'setValidator(address[] _validator, bool[] _val)'
+  readonly setIsBatchPosterFn =
+    'setIsBatchPoster(address addr, bool isBatchPoster_)'
+  readonly ownerFunctionCalledEvent = 'OwnerFunctionCalled(uint256 indexed id)'
+  readonly interface = new utils.Interface([
+    `function ${this.setValidatorFn}`,
+    `function ${this.setIsBatchPosterFn}`,
+    `event ${this.ownerFunctionCalledEvent}`,
+  ])
+  readonly setValidatorSighash = this.interface.getSighash(this.setValidatorFn)
+  readonly setIsBatchPosterSighash = this.interface.getSighash(
+    this.setIsBatchPosterFn,
+  )
+
+  constructor(
+    readonly field: string,
+    readonly definition: ArbitrumActorsHandlerDefinition,
+    readonly logger: DiscoveryLogger,
+  ) {}
+
+  async execute(
+    provider: DiscoveryProvider,
+    address: EthereumAddress,
+    blockNumber: number,
+  ): Promise<HandlerResult> {
+    this.logger.logExecution(this.field, [
+      this.definition.actorType === 'validator'
+        ? 'Fetching Arbitrum Validators'
+        : 'Fetching Arbitrum Batch Posters',
+    ])
+
+    // Find transactions in which setValidator/setIsBatchPoster was called
+    const logs = await this.getRelevantLogs(provider, address, blockNumber)
+    const txHashes = logs.map((log) => Hash256(log.transactionHash))
+
+    // Extract setValidator/setIsBatchPoster call parameters
+    // from transaction traces and process them
+    const isActor: Record<string, boolean> = {}
+    for (const txHash of txHashes) {
+      const traces = await provider.getTransactionTrace(txHash)
```

Comment:
Please leave a comment here mentioning the fact that there is a potential speed-up when switching to `Promise.all`. It will be easier to find this place if we are ever searching for places that slow discovery down. I don't think we should move this to `Promise.all` right now since it won't make many calls at the moment.

Reply:
I wouldn't suggest to use Promise.all here given we don't know how many calls will be made here (could be hundreds in theory). Such places are quite easy to find anyway by searching for `for` loops with `await`s inside.